types/accesscontrol/access_operation_map.go

```diff
@@ -0,0 +1,23 @@
+package accesscontrol
+
+// Alias for Map of MessageIndex -> AccessOperation -> Channel
+type MessageAccessOpsChannelMapping = map[int]AccessOpsChannelMapping
+
+// Alias for Map of AccessOperation -> Channel
+type AccessOpsChannelMapping = map[*AccessOperation][]chan interface{}
+
+func WaitForSignals(accessOpsToChannelsMap AccessOpsChannelMapping) {
+	for _, channels := range accessOpsToChannelsMap {
+		for _, channel := range channels {
+			<-channel
+		}
+	}
+}
+
+func SendSignals(accessOpsToChannelsMap AccessOpsChannelMapping) {
```

Comment:
Let's call this something like SendAllSignals since we'll likely have one later that also takes a specific access op to signal for

Reply:
sg, updated th other one to also be called waitforallsignals